### DISCUSSION TITLE
Optional enum not shown in OpenAPI schema

### DISCUSSION THREAD

**Discussion by aidan-pf** [fastapi/fastapi#10316] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum
from typing import Optional

from fastapi import Depends, FastAPI, Request
from pydantic import BaseModel, Field

app = FastAPI()


class SortByOptions(Enum):
    top = "top"
    recent = "recent"


class InputParams(BaseModel):
    sort_by: Optional[SortByOptions] = Field(default=None, description="Sort by")
    # ... more optional enums here


@app.get("/sort-by")
async def sort_by(
    request: Request,
    sort_config: InputParams = Depends(),
):
    pass
```

### Description

* Optional enums do not appear in the generated docs as drop downs.

<img width="557" alt="Screenshot 2023-09-24 at 1 35 09 PM" src="https://github.com/tiangolo/fastapi/assets/95979452/0c7ed46b-8ce6-44ff-9efb-12ceb3f84643">            

* Desired behavior:       
          
<img width="420" alt="Screenshot 2023-09-24 at 1 35 51 PM" src="https://github.com/tiangolo/fastapi/assets/95979452/08192a4c-86d7-42fb-8522-41ecb0fd25f6">

* The only way I can get them to appear is by customizing the schema with:

```
@app.on_event("startup")
async def customize_openapi():
    openapi_schema = app.openapi()

    # Update this line with the correct path
    sort_route = openapi_schema["paths"]["/sort-by"]["get"]

    # Customize enum query parameters
    for param in sort_route.get("parameters", []):
        if param["name"] == "sort_by":
            param["schema"] = {
                "enum": [e.value for e in SortByOptions],
                "type": "string",
            }

    app.openapi_schema = openapi_schema
```

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.11.4

**Comment by maesapasap** [comment, marked as answer]
I encountered this as well. Found a related discussion: https://github.com/tiangolo/fastapi/discussions/9900, and workaround using pydantic: https://github.com/pydantic/pydantic/pull/6653

I assume you can do something similar, like
```python
sort_by: SortByOptions | SkipJsonSchema[None]
```

**Reply by aidan-pf** [reply]
That seems to help somewhat, by doing something like:
```
sort_by: Annotated[
        SearchSortByOptions,
        Field(
            description="Determines the sort order of search results to return.",
        ),
    ] | SkipJsonSchema[None] = None
```

However, the description "Determines the sort order of search results to return." still does not populate to swagger ui/fastapi docs.

**Reply by maesapasap** [reply]
Yeah, seems like a bug to me. My guess is it's caused by migrating to Pydantic v2. In the meantime, if you really want to show the description, you could add the fields directly to the route function, or use functions as dependencies


```python
class InputParams(BaseModel):
    sort_by: SortByOptions | None = None
    # ... more optional enums here


def _get_params(
    sort_by: Annotated[
        SortByOptions | SkipJsonSchema[None],
        Query(
            description="Determines the sort order of search results to return.",
        ),
    ] = None
    # ... other params
) -> InputParams:
    return InputParams(sort_by=sort_by)


@router.get("/sort-by")
async def sort_by(
    request: Request,
    # by function dependency
    sort_config: Annotated[InputParams, Depends(_get_params)],
    # or directly add the fields here
    sort_by_2: Annotated[
        SortByOptions | SkipJsonSchema[None],
        Query(
            description="Determines the sort order of search results to return.",
        ),
    ] = None,
) -> None:
    pass
```

**Comment by sharuman** [comment]
I solved it by using the `Query` class from fastapi:

```
from fastapi import Query

class InputParams(BaseModel):
    sort_by: SortByOptions = Query(None),
    # ... more optional enums here
```
